src/Microsoft.Performance.SDK/Processing/Options/IProcessingOptionsResolver.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Performance.SDK.Processing.DataSourceGrouping;
+using System.Collections.Generic;
+
+namespace Microsoft.Performance.SDK.Processing.Options
+{
+    public interface IProcessingOptionsResolver
+    {
+        /// <summary>
+        ///     Get the processor options for a group of data sources processed by a given processor.
+        /// </summary>
+        /// <param name="dataSourceGroup"> A group of data sources </param>
+        /// <param name="processingSource"> a processing source to process the data sources. </param>
+        /// <returns>Options <see cref="ProcessorOptions"/> to pass to the <see cref="IProcessingSource"/>.</returns>
+        ProcessorOptions GetProcessorOptions(IDataSourceGroup dataSourceGroup, IProcessingSource processingSource);
```

Comment:
I'm not sure passing `IProcessingSource` is going to work here, especially since `IProcessingSource` does not expose a `Guid`. We also don't want the implementors to mess around with the `IProcessingSource` instances that are being used by the engine to process things. This is both exposing too much and not enough at the same time 😆 

The [IDataSourceGroupResolver](https://github.com/microsoft/microsoft-performance-toolkit-sdk/blob/65348731b57f10e0d8d60b44f236676f3ce30d4d/src/Microsoft.Performance.SDK/Processing/DataSourceGrouping/IDataSourceGroupResolver.cs) interface passes in the `Guid` of the processing source. For consistency, we can do that.

Also for consistency, we should keep the position of the parameters the same (i.e. both have the guid first or second). Do you mind updating either this or `IDataSourceGroupResolver` so that they match?